torch_xla/csrc/batch_norm.cpp

```diff
@@ -26,31 +40,58 @@ xla::XlaOp BatchNormVarianceInvert(xla::XlaOp variance, float eps_value) {
 
 BatchNormOutput BuildBatchNormTraining(xla::XlaOp input, xla::XlaOp weight,
                                        xla::XlaOp bias, float eps_value) {
+  bool is_batchnorm_with_fp16_inputs =
+      IsF32BatchNormWithFP16Inputs(input, weight);
+  if (is_batchnorm_with_fp16_inputs) {
+    input = xla::ConvertElementType(input, xla::PrimitiveType::F32);
+  }
```

Comment:
Could you add a comment here to explain it is the amp use case, otherwise it might be a bit puzzling for TPU user. Same for other functions that does the convert.

Reply:
Done.